channeldb/channel.go

```diff
@@ -2156,74 +2152,6 @@ func deserializeLogUpdate(r io.Reader) (*LogUpdate, error) {
 	return l, nil
 }
 
-// CircuitKey is used by a channel to uniquely identify the HTLCs it receives
-// from the switch, and is used to purge our in-memory state of HTLCs that have
-// already been processed by a link. Two list of CircuitKeys are included in
-// each CommitDiff to allow a link to determine which in-memory htlcs directed
-// the opening and closing of circuits in the switch's circuit map.
-type CircuitKey struct {
```

Comment:
That's right. The first one being in the third commit. The invoices package uses `CircuitKey` which is defined in `channeldb`. If we leave it there, it would be impossible to import `invoices` from `channeldb` because `invoices` would have a hard dependency on `channeldb`.

Right now channeldb is a "mix" package with type definitions used across many packages, implementing db logic + some "business" logic like entity validations. The idea is to move those type definitions to the right packages but for the ones that are not clear (`CircuitKey` for example) have them in a subpackage that is imported by all the other packages that need it.

`CircuitKey` sounds like something that should be defined in the `htlc`-related package. But we can find its final home in future PRs